### ISSUE TITLE
Calendar Month and Year view multi-modal keyboard navigation doesn't work due to focus handling

### ISSUE THREAD

**Issue by damyanpetev** [IgniteUI/igniteui-angular#14921]
## Description  
Can't continue with KB navigation after clicking inside a Calendar Month or Year view.

 * igniteui-angular version: 18.1.x at least
 * browser: N/A

## Steps to reproduce  

1. Open https://www.infragistics.com/products/ignite-ui-angular/angular/components/calendar#angular-calendar-views
  OR https://stackblitz.com/edit/ecaxc2 (Month View cached fork)
  OR https://stackblitz.com/edit/nuyftt (Year View cached fork)
2. Click on any month/year item
3. Attempt to continue navigation with KB arrows for example

## Result  
KB nav doesn't work because the month/year view doesn't gain focus. Best guess atm there's a `mousedown` handler that prevents default so the wrapper container with `tabindex="0"` won't gain focus automatically and no logic backfills this behavior. If you click in-between items the view gains focus, which seems like a confirmation.

## Expected result  
The view should gain focus when interacted with pointer and allow navigation to continue with KB.

## Attachments  
Attach a sample if available, and screenshots, if applicable.
